test/test_torch.py

```diff
@@ -8409,29 +8409,33 @@ def test_scatter_add_bool(self):
 
     def test_masked_scatter(self):
         with warnings.catch_warnings(record=True) as w:
-            for dtype in [torch.uint8, torch.bool]:
-                num_copy, num_dest = 3, 10
-                dest = torch.randn(num_dest)
-                src = torch.randn(num_copy)
-                mask = torch.tensor((0, 0, 0, 0, 1, 0, 1, 0, 1, 0), dtype=dtype)
-                dest2 = dest.clone()
-                dest.masked_scatter_(mask, src)
-                j = 0
-                for i in range(num_dest):
-                    if mask[i]:
-                        dest2[i] = src[j]
-                        j += 1
-                self.assertEqual(dest, dest2, 0)
-
-                # make source bigger than number of 1s in mask
-                src = torch.randn(num_dest)
-                dest.masked_scatter_(mask, src)
-
-                # make src smaller. this should fail
-                src = torch.randn(num_copy - 1)
-                with self.assertRaises(RuntimeError):
+            for maskType in [torch.uint8, torch.bool]:
+                for dt in torch.testing.get_all_dtypes():
+                    if dt == torch.half or dt == torch.bool:
+                        continue
```

Comment:
first, do half and double not actually work for masking?

second, as has been mentioned in a number of reviews, don't skip tests, assert they don't work with a message saying to enable once it works.  As you've done it here, we will never actually run these tests, even when we enable bool and half.

Reply:
double?

Reply:
(probably meant bool)